src/AutoRest.CSharp/Common/Output/Models/Types/ModelTypeProvider.cs

```diff
@@ -4,109 +4,347 @@
+                null,
+                MethodSignatureModifiers.Internal,
+                fullParameterList,
+                new(true, GetBaseObjectType()?.InitializationConstructor.Signature.Parameters ?? Array.Empty<Parameter>()));
         }
 
-        private static ConstructorSignature CreatePublicConstructor(string name, InputModelTypeUsage usage, IEnumerable<Parameter> parameters)
+        private ConstructorSignature CreatePublicConstructorSignature(string name, InputModelTypeUsage usage, IEnumerable<Parameter> parameters)
         {
+            //get base public ctor params
+            List<Parameter> fullParameterList = new List<Parameter>();
+            var parent = GetBaseObjectType();
+            if (parent is not null && parent.InitializationConstructor is not null)
+            {
+                fullParameterList.AddRange(parent.InitializationConstructor.Signature.Parameters);
+            }
+            fullParameterList.AddRange(parameters.Select(CreatePublicConstructorParameter));
+
             var summary = $"Initializes a new instance of {name}";
             var accessibility = usage == InputModelTypeUsage.Output
                 ? MethodSignatureModifiers.Internal
                 : MethodSignatureModifiers.Public;
 
-            return new ConstructorSignature(name, summary, null, accessibility, parameters.Select(CreatePublicConstructorParameter).ToList());
+            return new ConstructorSignature(
+                name,
+                summary,
+                null,
+                accessibility,
+                fullParameterList,
+                new(true, GetBaseObjectType()?.InitializationConstructor.Signature.Parameters ?? Array.Empty<Parameter>()));
         }
 
         private static Parameter CreatePublicConstructorParameter(Parameter p)
             => TypeFactory.IsList(p.Type) ? p with { Type = new CSharpType(typeof(IEnumerable<>), p.Type.IsNullable, p.Type.Arguments) } : p;
 
         private static Parameter CreateSerializationConstructorParameter(Parameter p) // we don't validate parameters for serialization constructor
             => p with { Validation = ValidationType.None };
+
+        protected override ObjectTypeConstructor BuildInitializationConstructor()
+        {
+            ObjectTypeConstructor? baseCtor = GetBaseObjectType()?.InitializationConstructor;
+
+            return new ObjectTypeConstructor(InitializationConstructorSignature, GetPropertyInitializers(Fields.PublicConstructorParameters), baseCtor);
+        }
+
+        protected override ObjectTypeConstructor BuildSerializationConstructor()
+        {
+            ObjectTypeConstructor? baseCtor = GetBaseObjectType()?.SerializationConstructor;
+
+            return new ObjectTypeConstructor(SerializationConstructorSignature, GetPropertyInitializers(Fields.SerializationParameters), baseCtor);
+        }
+
+        private ObjectPropertyInitializer[] GetPropertyInitializers(IReadOnlyList<Parameter> parameters)
```

Comment:
Should this be `GetParameterInitializers`?

Reply:
Based on the type name I would leave it as is.  I also think property initializer is more correct.  We are using the parameters to initialize the properties, not sure what "parameter initialization" would mean.